File: Hactathon/royal-square-crm-springboot/src/main/java/za/co/royalsquare/crm/policy/Policy.java
package za.co.royalsquare.crm.policy;

import jakarta.persistence.*;
import za.co.royalsquare.crm.client.Client;
import za.co.royalsquare.crm.common.Auditable;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.util.UUID;

@Entity
@Table(name = "policy")
public class Policy extends Auditable {

    @Id
    @GeneratedValue
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "client_id")
    private Client client;

    /** Sanlam, Old Mutual, Liberty, Momentum, Discovery, Allan Gray, Santam. */
    @Column(nullable = false, length = 80)
    private String provider;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 40)
    private ProductType productType;

    @Column(nullable = false, length = 40)
    private String policyNumber;

    @Column(precision = 15, scale = 2)
    private BigDecimal sumAssured;

    @Column(precision = 15, scale = 2)
    private BigDecimal monthlyPremium;

    private LocalDate renewalDate;

    protected Policy() { }

    public Policy(String provider, ProductType type, String policyNumber) {
        this.provider = provider;
        this.productType = type;
        this.policyNumber = policyNumber;
    }

    public UUID getId() { return id; }
    public String getProvider() { return provider; }
    public ProductType getProductType() { return productType; }
    public String getPolicyNumber() { return policyNumber; }
    public BigDecimal getSumAssured() { return sumAssured; }
    public BigDecimal getMonthlyPremium() { return monthlyPremium; }
    public LocalDate getRenewalDate() { return renewalDate; }

    void setClient(Client client) { this.client = client; }
    public void setSumAssured(BigDecimal v) { this.sumAssured = v; }
    public void setMonthlyPremium(BigDecimal v) { this.monthlyPremium = v; }
    public void setRenewalDate(LocalDate d) { this.renewalDate = d; }
}

File: royal-square-crm-python/app/models/policy.py
import uuid
from sqlalchemy import Column, String, Numeric, Date, ForeignKey
from sqlalchemy.orm import relationship
from app.database import Base
def gen_uuid(): return str(uuid.uuid4())
class Policy(Base):
    __tablename__ = "policies"
    id = Column(String, primary_key=True, default=gen_uuid)
    client_id = Column(String, ForeignKey("clients.id"), nullable=False)
    provider = Column(String(80), nullable=False)
    product_type = Column(String(40), nullable=False)
    policy_number = Column(String(40), nullable=False)
    sum_assured = Column(Numeric(15,2), nullable=True)
    monthly_premium = Column(Numeric(15,2), nullable=True)
    renewal_date = Column(Date, nullable=True)
    client = relationship("Client", back_populates="policies")
